pkg/lock/lock_solaris.go

```diff
@@ -24,12 +24,8 @@ import (
 	"syscall"
 )
 
-// LockedOpenFile - initializes a new lock and protects
-// the file from concurrent access across mount points.
-// This implementation doesn't support all the open
-// flags and shouldn't be considered as replacement
-// for os.OpenFile().
-func LockedOpenFile(path string, flag int, perm os.FileMode) (*LockedFile, error) {
+// lockedOpenFile is an internal function.
+func lockedOpenFile(path string, flag int, perm os.FileMode, alock int) (*LockedFile, error) {
```

Comment:
1. `alock` name doesn't convey the meaning.
2. By default `var lock syscall.Flock_t` created fields with zero value.  Not sure why you initialize `Start`, `Len`, `Pid` explicitly.